### DISCUSSION TITLE
Relationship between Backlog size and Retention size

### DISCUSSION THREAD

**Discussion by sh-y00** [apache/pulsar#20191] [Q&A] [answered]
Can someone please tell me the relationship between Backlog size and Retention size, because I don't understand it.

●Assumption
Backlog Size：500GB

●Description
When I tried to set the Retention size to 100GB, the following error was output.
    _Retention Quota must exceed configured backlog quota for topic. Please increase retention quota and retry_
    _Reason: Retention Quota must exceed configured backlog quota for topic. Please increase retention quota and retry_

●My understanding
Backlog size and Retention size are areas used for different purposes, so there was no dependency between them.

●Question
①I would like to know if there is anything wrong about my understanding.
②I would like to know the relationship between Backlog size and Retention size.

**Comment by tisonkun** [comment, marked as answer]
Hi @sh-y00!

Given Retention size=100GB and Backlog Size=500GB, if your backlog exceeds 100GB, it will be dropped due to retention policy. Thus, there is no backlog beyond 100GB.

Instead of breaking the semantic of option, Pulsar fails fast to indicate a semantic error on configurations.
